src/project/internal/projectactionscontroller.cpp

```diff
@@ -930,7 +934,24 @@ bool ProjectActionsController::saveProjectLocally(const muse::io::path_t& filePa
 
     if (!ret) {
         LOGE() << ret.toString();
-        warnScoreCouldnotBeSaved(ret);
+        if (ret.code() != (int)Err::CorruptionUponSavingError) {
+            warnScoreCouldnotBeSaved(ret);
+        } else {
+            switch (warnScoreHasBecomeCorruptedAfterSave(ret)) {
+            case RETRY_SAVE_BTN_ID:
+                async::Async::call(this, [this, filePath, saveMode]() {
+                    const ActionData actionData = ActionData::make_arg2<muse::io::path_t, SaveMode>(filePath, saveMode);
+                    dispatcher()->dispatch("file-retry-save-locally", actionData);
+                });
+                break;
+
+            case SAVE_AS_BTN_ID:
+                async::Async::call(this, [this]() {
+                    dispatcher()->dispatch("file-save-as");
```

Comment:
Call saveProject(SaveMode::SaveAs) instead of sending the action. Also, async::Async::call is not needed here (same for the previous case)